polygon/bor/bor.go

```diff
@@ -307,6 +307,10 @@ func BorRLP(header *types.Header, c *borcfg.BorConfig) []byte {
 	return b.Bytes()
 }
 
+type spanProducersReader interface {
+	Producers(ctx context.Context, blockNum uint64) (*valset.ValidatorSet, error)
```

Comment:
as discussed offline, maybe no need for a 2nd interface, can attach this function to the existing `spanReader` interface since the 2 data entities are logically tied together